lib/sqlalchemy/dialects/mysql/reflection.py

```diff
@@ -18,31 +25,40 @@
 from ... import types as sqltypes
 from ... import util
 
+if TYPE_CHECKING:
+    from .base import MySQLDialect
+    from .base import MySQLIdentifierPreparer
+    from ...engine.interfaces import ReflectedColumn
+
 
 class ReflectedState:
     """Stores raw information about a SHOW CREATE TABLE statement."""
 
-    def __init__(self):
-        self.columns = []
-        self.table_options = {}
-        self.table_name = None
-        self.keys = []
-        self.fk_constraints = []
-        self.ck_constraints = []
+    def __init__(self) -> None:
+        self.columns: list[ReflectedColumn] = []
+        self.table_options: dict[str, str] = {}
+        self.table_name: Optional[str] = None
+        self.keys: list[dict[str, Any]] = []
+        self.fk_constraints: list[dict[str, Any]] = []
+        self.ck_constraints: list[dict[str, Any]] = []
 
 
 @log.class_logger
-class MySQLTableDefinitionParser:
+class MySQLTableDefinitionParser(log.Identified):
```

Comment:
It may be useful to explain in commit message (PR description) why the inheritance got changed.

Reply:
This is a typing PR, any change that's not insignificant should be taken out.